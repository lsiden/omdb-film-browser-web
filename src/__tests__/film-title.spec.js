import React from "react"
import { shallow } from "enzyme"

import FilmTitle from "film-title"

const films = require("./films.json").Search

const defaultProps = () => ({
  filmSummary: films[0],
})

const createWrapper = (props = {}) =>
  shallow(<FilmTitle {...{ ...defaultProps(), ...props }} />)

test("click on title invokes dispatchViewDetail()", () => {
  expect(createWrapper()).toMatchSnapshot()
})
