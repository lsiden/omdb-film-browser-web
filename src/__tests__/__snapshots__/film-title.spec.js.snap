// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click on title invokes dispatchViewDetail() 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmTitle
    filmSummary={
        Object {
            "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg",
            "Title": "The Hobbit: The Battle of the Five Armies",
            "Type": "movie",
            "Year": "2014",
            "imdbID": "tt2310332",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        style={
                Object {
                        "color": undefined,
                      }
        }
        title="View film details"
        to="/film/tt2310332"
>
        The Hobbit: The Battle of the Five Armies, 2014
</Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "The Hobbit: The Battle of the Five Armies, 2014",
        "replace": false,
        "style": Object {
          "color": undefined,
        },
        "title": "View film details",
        "to": "/film/tt2310332",
      },
      "ref": null,
      "rendered": "The Hobbit: The Battle of the Five Armies, 2014",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          style={
                    Object {
                              "color": undefined,
                            }
          }
          title="View film details"
          to="/film/tt2310332"
>
          The Hobbit: The Battle of the Five Armies, 2014
</Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "The Hobbit: The Battle of the Five Armies, 2014",
          "replace": false,
          "style": Object {
            "color": undefined,
          },
          "title": "View film details",
          "to": "/film/tt2310332",
        },
        "ref": null,
        "rendered": "The Hobbit: The Battle of the Five Armies, 2014",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
